src/transformers/integrations.py

```diff
@@ -87,14 +87,6 @@ def is_sigopt_available():
     return importlib.util.find_spec("sigopt") is not None
 
 
-def is_azureml_available():
```

Comment:
@wamartin-aml  I think it needs to be a bit more nuanced than this within a minor release window to make sure customers that only have the older core SDK aren't broken.

If azureml-mlflow is installed, it's a superset and the AML callback should be removed
If azureml-core is the only one installed and no MLflow, the AML callback should probably activated and mlflow deactivated

etc